Готовая прога(версия QA)

stlegion277 opened the pull request:
Вродь все

stlegion277 committed:
Дописал всю базу. Некоторые функции из UML откинул за ненадобностью. Потому что нет смысла делать это в задании без звезд.

stlegion277 committed:
Merge branch 'Alibek' of https://github.com/50ohmTech/SummerSweat into Alibek

stlegion277 committed:
Поправил замечания Феди и Андрея.

stlegion277 committed:
Добавил формы. (две недели назад)

stlegion277 committed:
Статичное добавление тестовых цепей и калькулятор.

stlegion277 committed:
Готовая прога(версия QA).

stlegion277 committed:
Пара фиксов.

stlegion277 committed:
Мердж девелопа

DmitryKozhin commented on Alibek/ElementsLibrary/Circuit.cs:
```diff
@@ -1,70 +1,103 @@
 {
     /// <summary>
-    /// Класс электрическая цепь Circuit
+    ///     Класс электрическая цепь Circuit <see cref="Circuit" />
```
`<see cref="Circuit" />` для чего подобные ссылки?

stlegion277 replied:
+

DmitryKozhin commented on Alibek/ElementsLibrary/Circuits/ParallelSubcircuit.cs:
```diff
@@ -0,0 +1,45 @@
+        /// <returns>Комплексное значение импеданса</returns>
+        public override Complex CalculateZ(double frequency)
+        {
+            if ((frequency <= 0) || (frequency > 1000000000000))
```
В константы

stlegion277 replied:
+

DmitryKozhin commented on Alibek/ElementsLibrary/Circuits/SubcircuitBase.cs:
```diff
@@ -0,0 +1,69 @@
+        /// <summary>
+        ///     Идентификатор
+        /// </summary>
+        public uint uniqueID { get; }
```
Неправильное именование

stlegion277 replied:
+

DmitryKozhin commented on Alibek/MainForm/CircuitForm.cs:
```diff
@@ -0,0 +1,332 @@
+            TestCircuitsComboBox.Items.Add("Цепь №3");
+            TestCircuitsComboBox.Items.Add("Цепь №4");
+            TestCircuitsComboBox.Items.Add("Цепь №5");
+            TestCircuitsComboBox.DropDownStyle = ComboBoxStyle.DropDownList;
```
Это лучше установить в .Designer.cs файле

stlegion277 replied:
+

DmitryKozhin commented on Alibek/MainForm/ImpedanceCalculator.cs:
```diff
@@ -0,0 +1,144 @@
+
+            var start = double.Parse(StartValueTextBox.Text);
+            var step = double.Parse(StepValueTextBox.Text);
+            var finish = uint.Parse(EndValueTextBox.Text);
```
Тут может быть exception. Используй double.TryParse или try/catch (лучше первое)

stlegion277 replied:
+

DmitryKozhin commented on Alibek/MainForm/ValueValidators.cs:
```diff
@@ -0,0 +1,186 @@
+        public static void PressInt(object sender, KeyPressEventArgs e)
+        {
+            if (e.KeyChar == (char) Keys.Back || char.IsDigit(e.KeyChar) &&
+                ((TextBox) sender).Text.Length < 4)
```
4 в константу

stlegion277 replied:
+

DmitryKozhin commented on Alibek/MainForm/ImpedanceCalculator.cs:
```diff
@@ -0,0 +1,144 @@
+            StartValueTextBox.ContextMenu = new ContextMenu();
+            StepValueTextBox.ContextMenu = new ContextMenu();
+            EndValueTextBox.ContextMenu = new ContextMenu();
+            _circuit = circuit;
```
circuit может быть null?

DmitryKozhin commented on Alibek/MainForm/CircuitForm.cs:
```diff
@@ -0,0 +1,332 @@
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void treeView_MouseDoubleClick(object sender, MouseEventArgs e)
```
Именование методов должно начинаться с большой буквы

stlegion277 replied:
+

stlegion277 committed:
Поправил замечания.

stlegion277 committed:
Поправил формы.

YulyaNovichkova commented:
Пустой комбобокс, "Выберите цепь" не отображается по дефолту.

YulyaNovichkova commented:
Ещё при изменении элемента в новой форме не хватает label'а, который бы говорил том, мол "Введите новое значение": или просто "Новое значение:"

stlegion277 committed:
Поправил замечания Юли.

stlegion277 committed:
Добавил XML комментарии.

UncleThed commented on Alibek/ElementsLibrary/Circuit.cs:
```diff
@@ -1,70 +1,104 @@
+        /// </summary>
+        /// <param name="currentNode">Текущая нода</param>
+        /// <param name="newNode">Новая нода</param>
+        public void AddAfter(INode currentNode, INode newNode)
```
newNode == null

stlegion277 replied:
+

UncleThed commented on Alibek/MainForm/CircuitForm.cs:
```diff
@@ -0,0 +1,337 @@
+            var selectedCircuit = testCircuitsComboBox.SelectedItem.ToString();
+            if (selectedCircuit == "Цепь №1")
+            {
+                ElementBase resistor = new Resistor(13, "R1");
```
По моему, не стоит менять порядок параметров для одного типа элементов

stlegion277 replied:
+

stlegion277 committed:
Поправил замечания Юли и Феди.

Nachyn reviewed (changes requested):
в ЛС отправил замечания

stlegion277 committed:
Поправил замечания Феди и Начына.

Commit:
Добавлены замечания по коду.
Код скопирован у Шагаева - всё скопированное удалить и написать своё. Добавить события

goryainovalex commented:
Оставил замечания в коммите (см. TODO)

stlegion277 committed:
Поправил замечания Начына

stlegion277 committed:
Merge branch 'Alibek' of https://github.com/50ohmTech/SummerSweat into Alibek

stlegion277 committed:
Часть правок. (промежуточный коммит)

stlegion277 committed:
Поправил замечания, переписал стыдобу.

Nachyn reviewed (approved):
ИСПРАВЬ ЗАМЕЧАНИЯ

stlegion277 committed:
Исправил замечания Начына.

stlegion277 committed:
Правки

Commit:
Еще замечания

stlegion277 committed:
Исправил еще правки.

stlegion277 committed:
Исправил правки, теперь пришли новые правки.

stlegion277 committed:
Промежуточный коммит
